scio-google-cloud-platform/src/main/scala/com/spotify/scio/pubsub/PubsubIO.scala

```diff
@@ -108,28 +111,40 @@ object PubsubIO {
   ): PubsubIO[(T, Map[String, String])] =
     PubsubIOWithAttributes[T](name, idAttribute, timestampAttribute)
 
-  private[pubsub] def setAttrs[T](
-    r: beam.PubsubIO.Read[T]
-  )(idAttribute: String, timestampAttribute: String): beam.PubsubIO.Read[T] = {
-    val r0 = Option(idAttribute)
-      .map(att => r.withIdAttribute(att))
-      .getOrElse(r)
+  private[pubsub] def configureRead[T](
+    read: beam.PubsubIO.Read[T]
+  )(
+    name: String,
+    params: ReadParam,
+    idAttribute: String,
+    timestampAttribute: String
+  ): beam.PubsubIO.Read[T] = {
+    val r0 =
+      params.readType match {
+        case Subscription => read.fromSubscription(name)
+        case Topic        => read.fromTopic(name)
+      }
 
-    Option(timestampAttribute)
-      .map(att => r0.withTimestampAttribute(att))
-      .getOrElse(r0)
+    val r1 = params.clientFactory.fold(r0)(r0.withClientFactory)
+    val r2 = params.deadLetterTopic.fold(r1)(r1.withDeadLetterTopic)
+    val r3 = Option(idAttribute).fold(r2)(r2.withIdAttribute)
+    Option(timestampAttribute).fold(r3)(r3.withTimestampAttribute)
   }
 
-  private[pubsub] def setAttrs[T](
-    r: beam.PubsubIO.Write[T]
-  )(idAttribute: String, timestampAttribute: String): beam.PubsubIO.Write[T] = {
-    val r0 = Option(idAttribute)
-      .map(att => r.withIdAttribute(att))
-      .getOrElse(r)
-
-    Option(timestampAttribute)
-      .map(att => r0.withTimestampAttribute(att))
-      .getOrElse(r0)
+  private[pubsub] def configureWrite[T](
+    write: beam.PubsubIO.Write[T]
+  )(
+    name: String,
+    params: WriteParam,
+    idAttribute: String,
+    timestampAttribute: String
+  ): beam.PubsubIO.Write[T] = {
+    val w0 = write.to(name)
```

Comment:
thanks for catching the bug with `.with`! Not sure how that one snuck in there.

Usually in our IOs we use a pattern of `var` + `fold` to apply successive `.withX...` statements, like this: https://github.com/spotify/scio/blob/main/scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala#L84-L89

Reply:
Fixed!